Source/SwiftLintFramework/Helpers/RegexHelpers.swift

```diff
@@ -8,7 +8,7 @@
 
 struct RegexHelpers {
     // A single variable
-    static let varName = "[a-zA-Z_][a-zA-Z0-9_]+"
+    static let varName = "[a-zA-Z_\.][a-zA-Z0-9_\.]+"
```

Comment:
If the initial dot is for extensions, then clearly declare it as an optional prefix:

    "\.?[a-zA-Z_][a-zA-Z0-9_\.]+"

Also you will want to document your pattern as lenient, as it doesn't catch invalid cases of consecutive dots or final dots.